models/gif_psc_exp.h

```diff
@@ -0,0 +1,425 @@
+
+    /** Refractory period in ms. */
+    double_t t_ref_;
+
+    /** Membrane capacitance in pF. */
+    double_t c_m_;
+
+    /** List of spike triggered current time constant in ms. */
+    std::vector< double_t > tau_stc_;
+
+    /** List of spike triggered current jumps in nA. */
+    std::vector< double_t > q_stc_;
+
+    /** List of adaptive threshold time constant in ms. */
+    std::vector< double_t > tau_sfa_;
+
+    /** List of adaptive threshold jumps in mV. */
+    std::vector< double_t > q_sfa_;
+
+    /** Time constant of excitatory synaptic current in ms. */
+    double_t tau_ex_;
+
+    /** Time constant of inhibitory synaptic current in ms. */
+    double_t tau_in_;
+
+
+    /** External DC current. */
+    double_t I_e_;
+
+    Parameters_(); //!< Sets default parameter values
+
+    void get( DictionaryDatum& ) const; //!< Store current values in dictionary
+    void set( const DictionaryDatum& ); //!< Set values from dictionary
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   */
+  struct State_
+  {
+    double_t y0_;  //!< This is piecewise constant external current
+    double_t y3_;  //!< This is the membrane potential RELATIVE TO RESTING POTENTIAL.
+    double_t q_;   //!< This is the change of the 'threshold' due to adaptation.
+    double_t stc_; // Spike triggered current.
+
+    std::vector< double_t > q_sfa_elems_; // Vector of adaptation parameters.
+    std::vector< double_t > q_stc_elems_; // Vector of spike triggered parameters.
+
+    double_t i_syn_ex_; // postsynaptic current for exc.
+    double_t i_syn_in_; // postsynaptic current for inh.
+
+    int_t r_ref_; // absolute refractory counter (no membrane potential propagation)
+
+    bool initialized_; // it is true if the vectors are initialized
+    bool add_stc_sfa_; // in case of true, the stc and sfa ampplitudes should be
+                       // added
+
+    State_(); //!< Default initialization
+
+    void get( DictionaryDatum&, const Parameters_& ) const;
+    void set( const DictionaryDatum&, const Parameters_& );
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Buffers of the model.
+   */
+  struct Buffers_
+  {
+    Buffers_( gif_psc_exp& );
+    Buffers_( const Buffers_&, gif_psc_exp& );
+
+    /** buffers and sums up incoming spikes/currents */
+    RingBuffer spikes_ex_;
+    RingBuffer spikes_in_;
+    RingBuffer currents_;
+
+    //! Logger for all analog data
+    UniversalDataLogger< gif_psc_exp > logger_;
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Internal variables of the model.
+   */
+  struct Variables_
+  {
+
+    double_t P30_;
+    double_t P33_;
+    double_t P31_;
+    double_t P11ex_;
+    double_t P11in_;
+    double_t P21ex_;
+    double_t P21in_;
+    std::vector< double_t > Q33_; // for sfa
+    std::vector< double_t > Q44_; // for stc
```

Comment:
I think `P_sfa_` and `P_stc_` would be more useful names, making clear to which quantities these propagator elements apply.